### PR TITLE
Allow editing an upcoming match

### PR DESCRIPTION
**Pull request by harrygr**

### CODE REVIEW

File: src/pages/Match/EditMatch.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                      Go back
+                    </ButtonLink>
+                  </Vspace>
+                )
+              ) : (
+                <p>
+                  You're not allowed to edit this match 🚫. Are you the owner?{' '}
+                  <Link to="/login">Login</Link> to edit it.
+                </p>
+              )}
```
[the remark this answers is not in the document]
Comment (Jarrydcrawford) [reply src/pages/Match/EditMatch.tsx]: Just about 😏

File: src/pages/Match/EditMatch.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            form,
+            actions,
+          ) => {
+            const formEventMeta = { form: 'update_match' };
+            return updateMatch({
+              variables: {
+                id: match.id,
+                input: sanitizeUpdateMatchArgs(form),
+              },
+            }).then(r => {
```
Comment (Jarrydcrawford) [inline src/pages/Match/EditMatch.tsx]: Avoid single character variable names if we can help it? 😖

File: src/pages/Match/matchUtils.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export const sanitizeUpdateMatchArgs = (
+  values: MatchFields,
+): UpdateMatchInput => {
+  return {
+    opponent: values.opponent,
+    playedAt: makeMatchDate(values.matchDate, values.matchTime),
+  };
+};
+
+export const ownsMatch = (match: Match) => (user: User) => {
```
Comment (Jarrydcrawford) [inline src/pages/Match/matchUtils.ts]: neat

File: src/pages/Match/EditMatch.tsx (hunk 2 above)
Comment (harrygr) [reply src/pages/Match/EditMatch.tsx]: yeah, I figured there's so much nesting the apollo response it keeps things a bit less verbose. Can change it though

**Pull request merged**